### ISSUE TITLE
Using RTK Query with an existing Redux store that uses sagas?

### ISSUE THREAD

**Issue by 00e6** [reduxjs/redux-toolkit#3753]
Hi,
Our current app uses Redux Sagas. But we end up writing a lot of boiler plate for what is normally just a simple GET or POST request. 

I'm trying to migrate to RTK Query, but having a lot of issues calling configureStore with the RTK Query api (from `createApi()`) and sagas. 

I'm using NextJS. I can get it working in development mode but when building (for production / deployment) I get `TypeError: Cannot read properties of undefined (reading 'api')` errors (from `@reduxjs/toolkit/dist/query/rtk-query.esm.js`).

in my main app.tsx I have redux provider as a parent correctly. 
The error only occurs one i use a hook (e.g. `useGetSomeDataQuery`) in a component. It is mostly wired up correctly as i can get it working in a browser in dev mode.

This is the rough outline of current code:

```tsx
// simplified for github issue but this is the general idea
const createReducer = (injectedReducers: InjectedReducersType = {}) => {
  const keys = Object.keys(injectedReducers);
  if (keys.length === 0   ) {
    return (state) => state;
  } 

    return combineReducers({
      ...injectedReducers,
      [rktQueryApi.reducerPath]: rktQueryApi.reducer, 
    });
  
};



  const sagaMiddleware = createSagaMiddleware();
  const runSaga  = sagaMiddleware.run

  const middlewares = [sagaMiddleware];
  const sagaEnhancer =   createInjectorsEnhancer({
      createReducer,
      runSaga,
    })

  const enhancers = [  sagaEnhancer  ];

  const rootReducer = createReducer({
    [rktQueryApi.reducerPath]: rktQueryApi.reducer,
  });

  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(rktQueryApi.middleware, ...middlewares),
    enhancers: enhancers,
  });


```

I think its due to createReducer (either in the enhancer or in initial setup) but I can't figure out how to get it to create the reducer correctly. Is there something obvious I am missing here?

I can't find much (if any) documentation for using both sagas + RTK query. Eventually I'd like to move it entirely to RTK Query but the app is too large to do that in one big go.

thanks!

**Comment by markerikson** [comment]
One issue is that you're blowing away the existing enhancers setup by supplying an entire array for the `enhancers` arg.

Instead, try: 

``ts
enhancers: (defaultEnhancers) => {
  return [...defaultEnhancers, sagaEnhancer]
}
```

(But also we would strongly recommend moving away from sagas for data fetching ASAP!)

**Comment by 00e6** [comment]
oh .. that was it

I spent the last couple of hours trying to debug this. thank you very much @markerikson 

(It was a combination of both things you suggested: the defaultEnhancers, and removing the conditional thing in createReducer)

I think that conditional code was from some boilerplate/starter code (I can find it on a few results on Google e.g. [here](https://stackoverflow.com/questions/67140586/react-connect-react-router-could-not-find-router-reducer-in-state-tree-it-mus)).... I thought I tried it without the condition (I couldn't really understand why we needed that empty state...) - but i think it needed the defaultEnhancers change as well.

> (But also we would strongly recommend moving away from sagas for data fetching ASAP!)

Agree with, and am working on moving away from sagas. RTK Query is much cleaner code (i've replaced lots of saga code with just a few lines in createApi() )
